decorator.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'nameable'
+
+class Decorator
+  include Nameable
+
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super[0...10]
```

Comment:
- [ ] [OPTIONAL] You can improve your code by only trimming the name if it has more than 10 characters e.g. `@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`